src/chttpd/src/chttpd.erl

```diff
@@ -1045,6 +1051,39 @@ send_error(Req, Code, Headers, ErrorStr, ReasonStr, Stack) ->
         case Stack of [] -> []; _ -> [{<<"ref">>, stack_hash(Stack)}] end
     ]}).
 
+update_timeout_stats(Error, PathParts)
```

Comment:
General rambling...

I don't like that we are exposing endpoint paths in `chttpd.erl`. I know we already do request specific handling in `chttpd:error_info`. This makes implementation of an endpoint coupled with correct update in `chttpd`. I would rather prefer not to leak this detail. However I don't see an easy way to achieve that without changing the way how http handlers are plugged. Changing that aspect is clearly out of scope of the PR. 

Back to this PR...
I think we should convert this function to use function clauses. It might look something like:
```
update_timeout_stats(<<"timeout">>, #httpd{requested_path_parts = RequestedPathParts} = Req) ->
   update_timeout_stats(RequestedPathParts);
update_timeout_stats(timeout, #httpd{requested_path_parts = RequestedPathParts} = Req) ->
   update_timeout_stats(RequestedPathParts).

update_timeout_stats([_, <<"_partition">>, _, <<"_design">>, _, <<"_view">> | _]) ->
     couch_stats:increment_counter([couchdb, httpd, partition_view_timeouts]);
...
```